kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/source/distribution/PartitionSearcher.scala

```diff
@@ -21,36 +21,30 @@ import com.typesafe.scalalogging.LazyLogging
 import io.lenses.streamreactor.connect.aws.s3.config.ConnectorTaskId
 import io.lenses.streamreactor.connect.aws.s3.model.location.S3Location
 import io.lenses.streamreactor.connect.aws.s3.source.config.PartitionSearcherOptions
-import io.lenses.streamreactor.connect.aws.s3.storage.CompletedDirectoryFindResults
+import io.lenses.streamreactor.connect.aws.s3.storage.AwsS3DirectoryLister
 import io.lenses.streamreactor.connect.aws.s3.storage.DirectoryFindCompletionConfig
-import io.lenses.streamreactor.connect.aws.s3.storage.PausedDirectoryFindResults
-import io.lenses.streamreactor.connect.aws.s3.storage.StorageInterface
-
-import cats.effect.Clock
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Request
+import software.amazon.awssdk.services.s3.model.ListObjectsV2Response
 
 class PartitionSearcher(
-  roots:    Seq[S3Location],
-  settings: PartitionSearcherOptions,
-)(
-  implicit
-  connectorTaskId:  ConnectorTaskId,
-  storageInterface: StorageInterface,
+  roots:           Seq[S3Location],
+  settings:        PartitionSearcherOptions,
+  connectorTaskId: ConnectorTaskId,
+  listS3ObjF:      ListObjectsV2Request => Iterator[ListObjectsV2Response],
 ) extends LazyLogging {
 
   def findNewPartitions(
     lastFound: Seq[PartitionSearcherResponse],
   ): IO[Seq[PartitionSearcherResponse]] =
     if (lastFound.isEmpty && roots.nonEmpty) {
-      roots.traverse(findNewPartitionsInRoot(_, settings, Set.empty, Option.empty, settings.clock))
+      roots.traverse(findNewPartitionsInRoot(_, settings, Set.empty))
```

Comment:
Not part of this PR. Using traverse here would make findNewPartitions fully fail for any single failing element inside root.

Reply:
That's a desired behaviour.